CONCEPTS.ja.md

```diff
@@ -297,12 +297,14 @@ ComputedModelで部分的にフィールドを定義したクラス (モジュ
 
 ```ruby
 module UserLikeConcern
-  extends ActiveSupport::Concern
+  extend ActiveSupport::Concern
   include ComputedModel::Model
 
-  dependency :preference, :profile
-  computed def display_name
-    "#{preference.title} #{profile.name}"
+  included do
```

Comment:
This is not an intended usage. Is there a problem with computed_model itself? If it's a pb-serializer's problem, please leave it as-is and write a notice somewhere else.

Reply:
Although `include do ... end` may work as well, this is unrelated to computed_model's inheritance feature, so the example won't fit here.

Reply:
This seems to be a problem on the computed model side.
Here is the code that reproduces it. If you do not use included, you will get an error.

```rb
module UserConcern
  extend ActiveSupport::Concern
  include ComputedModel::Model

  dependency raw_user: [:first_name, :last_name]
  computed def name
    [raw_user.first_name, raw_user.last_name].map(&:presence).compact.join(' ')
  end
end

# NoMethodError: undefined method `dependency' for UserConcern:Module
# from <main>:5:in `<module:UserConcern>'
```

Reply:
Thanks reproduced! It's strange though. Kinda `ActiveSupport::Concern`'s corner case? The following modifications work.

```ruby
module UserConcern
  # extend ActiveSupport::Concern
  include ComputedModel::Model

  dependency raw_user: [:first_name, :last_name]
  computed def name
    [raw_user.first_name, raw_user.last_name].map(&:presence).compact.join(' ')
  end
end
```

```ruby
module UserConcern
  extend ActiveSupport::Concern
  include ComputedModel::Model
  extend ComputedModel::Model::ClassMethods

  dependency raw_user: [:first_name, :last_name]
  computed def name
    [raw_user.first_name, raw_user.last_name].map(&:presence).compact.join(' ')
  end
end
```

Reply:
Anyway `include do ... end` is not an intended fix. Let's investigate it in https://github.com/wantedly/computed_model/issues/36